backend/schemas/sources/GeneralInformation.schema.jsonnet

```diff
@@ -1,148 +1,58 @@
+local Base = import 'Base.libsonnet';
+local Func = import 'Functions.libsonnet';
+local Types = Base.Types;
+
 {
    "$defs": {
-      "AuditPeriod": {
-         "description": "",
-         "enum": [
-            "annual",
-            "biennial",
-            "other"
-         ],
-         "title": "AuditPeriod",
-         "type": "string"
-      },
-      "EIN": {
+      "AuditPeriod": Base.Enum.AuditPeriod,
+      "EIN": Types.string_or_null {
          "pattern": "^[0-9]{9}$",
-         "type": [
-            "string",
-            "null"
-         ]
       },
-      "Phone": {
+      "Phone": Types.string {
          "pattern": "^^(\\+0?1\\s)?\\(?\\d{3}\\)?[\\s.-]?\\d{3}[\\s.-]?\\d{4}$",
-         "type": "string"
-      },
-      "State": {
-         "enum": [
-            "AL",
-            "AK",
-            "AZ",
-            "AR",
-            "CA",
-            "CO",
-            "CT",
-            "DE",
-            "FL",
-            "GA",
-            "HI",
-            "ID",
-            "IL",
-            "IN",
-            "IA",
-            "KS",
-            "KY",
-            "LA",
-            "ME",
-            "MD",
-            "MA",
-            "MI",
-            "MN",
-            "MS",
-            "MO",
-            "MT",
-            "NE",
-            "NV",
-            "NH",
-            "NJ",
-            "NM",
-            "NY",
-            "NC",
-            "ND",
-            "OH",
-            "OK",
-            "OR",
-            "PA",
-            "RI",
-            "SC",
-            "SD",
-            "TN",
-            "TX",
-            "UT",
-            "VT",
-            "VA",
-            "WA",
-            "WV",
-            "WI",
-            "WY"
-         ],
+      },
+      "State": Base.Enum.UnitedStatesStateAbbr {
          "title": "State",
-         "type": "string"
       },
-      "UEI": {
+      "UEI": Types.string_or_null {
```

Comment:
We'll catch this in a refactor; I have a more specific type, and we should probably not use a compound type as a base type. It buries optionality in a bad way. That is, if we want a field to be UEI_or_null, we should build that compound type, but not have the base type of UEI be optionally a string or null.